File: app/src/main/java/com/murrayde/animeking/model/Question.java
package com.murrayde.animeking.model;

import java.util.ArrayList;

public class Question {
    private String question;
    private String image_url;
    private String question_id;
    private ArrayList<String> multiple_choice;
    private int issue_count;
    private String anime_title;

    public String getQuestion() {
        return question;
    }

    public void setQuestion(String question) {
        this.question = question;
    }

    public String getImage_url() {
        return image_url;
    }

    public void setImage_url(String image_url) {
        this.image_url = image_url;
    }

    public String getQuestion_id() {
        return question_id;
    }

    public void setQuestion_id(String question_id) {
        this.question_id = question_id;
    }

    public ArrayList<String> getMultiple_choice() {
        return multiple_choice;
    }

    public void setMultiple_choice(ArrayList<String> multiple_choice) {
        this.multiple_choice = multiple_choice;
    }

    public int getIssue_count() {
        return issue_count;
    }

    public void setIssue_count(int issue_count) {
        this.issue_count = issue_count;
    }

    public String getAnime_title() {
        return anime_title;
    }

    public void setAnime_title(String anime_title) {
        this.anime_title = anime_title;
    }

    public Question() {
    }

    public Question(String question, String image_url, String question_id, ArrayList<String> multiple_choice, int issue_count, String anime_title) {
        this.question = question;
        this.image_url = image_url;
        this.question_id = question_id;
        this.multiple_choice = multiple_choice;
        this.issue_count = issue_count;
        this.anime_title = anime_title;
    }
}

File: app/src/main/java/com/murrayde/retrofittesting/model/Question.kt
@file:Suppress("FunctionName")

package com.murrayde.retrofittesting.model

data class Question(val question: String,
                    val image_url: String,
                    val question_id: Int,
                    val multiple_choice: ArrayList<String>,
                    val issue_count: Int,
                    val anime_title: String)

class QuestionBuilder {
    companion object {
        private val current_anime = HashMap<String, Int>()
        private var id_count = 0

        fun BUILD(question: String, image_url: String, multiple_choice: ArrayList<String>, issue_count: Int, anime_title: String): Question {
            if (current_anime.containsKey(anime_title)) {
                id_count = current_anime[anime_title]!!
                id_count++
                current_anime[anime_title] = id_count
            } else current_anime[anime_title] = 0
            return Question(question, image_url, current_anime.getValue(anime_title), multiple_choice, issue_count, anime_title)
        }
    }
}